qt/src/hocr/HOCRPdfExporter.cc

```diff
@@ -684,3 +740,41 @@ QImage HOCRPdfExporter::getSelection(const QRect& bbox)
 {
 	return m_displayerTool->getSelection(bbox);
 }
+
+void HOCRPdfExporter::paperSizeChanged() {
+	PaperSizeMetrics currentMetric = static_cast<PaperSizeMetrics>(ui.comboBoxPaperSizeMetric->itemData
+			(ui.comboBoxPaperSizeMetric->currentIndex()).toInt());
+
+	auto textPaperSize = ui.comboBoxPaperSize->itemData(ui.comboBoxPaperSize->currentIndex()).toString();
+	if(textPaperSize == "custom") {
+		ui.lineEditPaperSizeWidth->setEnabled(true);
+		ui.lineEditPaperSizeHeight->setEnabled(true);
+	} else if(textPaperSize == "source") {
+		ui.lineEditPaperSizeWidth->setText("");
+		ui.lineEditPaperSizeHeight->setText("");
+		ui.lineEditPaperSizeWidth->setEnabled(false);
+		ui.lineEditPaperSizeHeight->setEnabled(false);
+	} else {
+		ui.lineEditPaperSizeWidth->setEnabled(false);
+		ui.lineEditPaperSizeHeight->setEnabled(false);
+		QLocale locale;
+		QString widthText, heightText;
+		if(currentMetric == PaperSizeMetrics::cm) {
+			auto outputPaperSize = PaperSizeProcessor::getSizeMM(textPaperSize.toStdString());
+			widthText = locale.toString(static_cast<double>(outputPaperSize.width) / 10.0, 'f', 2);
+			heightText = locale.toString(static_cast<double>(outputPaperSize.height) / 10.0, 'f', 2);
+
+		} else if(currentMetric == PaperSizeMetrics::inch) {
+			auto outputPaperSize = PaperSizeProcessor::getSizeInch(textPaperSize.toStdString());
+			widthText = locale.toString(outputPaperSize.width, 'f', 2);
+			heightText = locale.toString(outputPaperSize.height, 'f', 2);
+		}
```

Comment:
You can save some code by just adding a `PaperSizeProcessor::getSize(unit)` function, moving the logic centrally to that function, instead needing the conditionals here and above in `OCRPdfExporter::run`.

Reply:
I think you meant these changes :-)